lib/jxl/encode_test.cc

```diff
@@ -1585,39 +1601,276 @@ TEST_P(EncodeOutputCallbackTest, OutputCallback) {
     if (p.use_container()) {
       JxlEncoderSetCodestreamLevel(enc, 10);
     }
-
-    EXPECT_EQ(JXL_ENC_SUCCESS,
-              JxlEncoderAddImageFrame(frame_settings, &frame.format,
-                                      frame.pixels(), frame.pixels_size));
-    if (p.multiple_frames()) {
+    for (size_t i = 0; i < number_extra_channels; i++) {
+      JxlExtraChannelInfo channel_info;
+      JxlExtraChannelType channel_type = JXL_CHANNEL_THERMAL;
+      JxlEncoderInitExtraChannelInfo(channel_type, &channel_info);
+      EXPECT_EQ(JXL_ENC_SUCCESS,
+                JxlEncoderSetExtraChannelInfo(enc, i, &channel_info));
+    }
+    if (add_image_frames) {
       EXPECT_EQ(JXL_ENC_SUCCESS,
                 JxlEncoderAddImageFrame(frame_settings, &frame.format,
                                         frame.pixels(), frame.pixels_size));
+      if (p.multiple_frames()) {
```

Comment:
you can use a for loop here to avoid duplicating the call

Reply:
done